sphinx/environment/__init__.py

```diff
@@ -758,6 +758,7 @@ def check_consistency(self) -> None:
                     continue
                 logger.warning(__("document isn't included in any toctree"),
                                location=docname)
+        _check_toc_parents(self.toctree_includes)
```

Comment:
I added the consistency check here instead of doing it while resolution in `_get_toctree_ancestors()` because `_get_toctree_ancestors()` is being called multiple times for the same document while writing.